percy/sitemap.js

```diff
@@ -0,0 +1,37 @@
+const Axios = require("axios")
+const PercyScript = require("@percy/script");
+const parseUrl = require("url-parse")
+const xml = require("xml2json")
+
+async function loadSitemap(url) {
+  const response = await Axios.get(url, {
+    responseType: "text",
+  })
+
+  return xml.toJson(response.data, {
+    object: true,
+  }).urlset.url
+}
+
+
+(async() => {
+
+  console.log('[INFO] Fetching sitemap');
+  const pageReferences = await loadSitemap(process.env.SITEMAP_URL)
+
+  console.log('[INFO] Found %d page references', pageReferences.length);
+  for (const i in pageReferences) {
+
+    const url = parseUrl(pageReferences[i].loc)
+    console.log('[INFO] Processing %s', url.href);
+
+    await PercyScript.run(async (page, percySnapshot) => {
+        await page.goto(url.href);
+        // ensure the page has loaded before capturing a snapshot
+        // await page.waitForSelector("body");
```

Comment:
@ekinoben what `waitForSelector` is for?

Reply:
The original example in Percy's docs included it - wait for a known element to be present before taking screenshot. But it seems - in my testing at least - that the screenshot is taken successfully anyway, so I've commented it out. It might be that we need to put it back in, or even add a variable where you can control which page element we should wait for before taking the screenshot.